packages/example-generator/src/run.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { Template } from "@osdk/create-app";
+import { run as runCreateApp, TEMPLATES } from "@osdk/create-app";
+import { consola } from "consola";
+import { compareSync } from "dir-compare";
+import { findUp } from "find-up";
+import { globby } from "globby";
+import { exec } from "node:child_process";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import { promisify } from "node:util";
+import * as tmp from "tmp";
+import { gitIgnoreFilter } from "./gitIgnoreFilter.js";
+
+interface RunArgs {
+  outputDirectory: string;
+  check: boolean;
+}
+
+interface Mutator {
+  filePattern: string;
+  mutate: (
+    template: Template,
+    existingContent: string,
+  ) => ModifyFile | DeleteFile;
+}
+
+interface ModifyFile {
+  type: "modify";
+  newContent: string;
+}
+
+interface DeleteFile {
+  type: "delete";
+}
+
+export async function run({ outputDirectory, check }: RunArgs) {
```

Comment:
It took me few reads to understand the flow. Perhaps a small refactor of dividing into few chunks could make it better? Possibly have the run function look like this:

```
 const tmpDir = setupTemporaryDirectory();

  await generateAndMutateProjects(tmpDir.name);

  await formatProjectsUsingMonorepolint(tmpDir.name);

  if (check) {
    await checkGeneratedProjects(outputPath, tmpDir.name);
  } else {
    copy(overwrite?)GeneratedProjects(outputPath, tmpDir.name);
  }```